pkg/protocol/engine/consensus/epochgadget/epochorchestrator/performance/testsuite_test.go

```diff
@@ -0,0 +1,188 @@
+package performance
+
+import (
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotaledger/hive.go/kvstore"
+	"github.com/iotaledger/hive.go/kvstore/mapdb"
+	"github.com/iotaledger/iota-core/pkg/core/account"
+	"github.com/iotaledger/iota-core/pkg/model"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/blocks"
+	"github.com/iotaledger/iota-core/pkg/storage/prunable"
+	"github.com/iotaledger/iota-core/pkg/utils"
+	iotago "github.com/iotaledger/iota.go/v4"
+	"github.com/iotaledger/iota.go/v4/tpkg"
+)
+
+type TestSuite struct {
+	T *testing.T
+
+	accounts map[string]iotago.AccountID
+
+	ProtocolParameters *iotago.ProtocolParameters
+
+	Instance *Tracker
+}
+
+func NewTestSuite(t *testing.T) *TestSuite {
+	ts := &TestSuite{
+		T:        t,
+		accounts: make(map[string]iotago.AccountID),
+
+		ProtocolParameters: &iotago.ProtocolParameters{
+			Version:     3,
+			NetworkName: utils.RandString(255),
+			Bech32HRP:   iotago.NetworkPrefix(utils.RandString(3)),
+			MinPoWScore: utils.RandUint32(50000),
+			RentStructure: iotago.RentStructure{
+				VByteCost:    100,
+				VBFactorData: 1,
+				VBFactorKey:  10,
+			},
+			TokenSupply:                      tpkg.RandBaseToken(math.MaxUint64),
+			GenesisUnixTimestamp:             time.Now().Unix(),
+			SlotDurationInSeconds:            10,
+			SlotsPerEpochExponent:            1,
+			ManaDecayFactors:                 []uint32{},
```

Comment:
I also wasn't sure about this, but even if we test with mana decay, then we need to provide the expected values, so we will use mana decay functions anyway. If we skip the decay we are sure that something is off directly with rewards calculations

Reply:
But if we never apply a mana decay, we don't test if we apply the correct decay in this logic.

If you simply use the mana decay provider to calculate the expected mana decay, then the result can't be wrong because of mana decay (even if the logic in the mana decay provider would be wrong).

I would recommend to add it.